### ISSUE TITLE
Array passed to `import` should not be modified

### ISSUE THREAD

**Issue by indirect** [zdennis/activerecord-import#203]
When calling `Model.import attrs, records`, the `attrs` array is modified—this is a highly unexpected side effect, and breaks if either `attrs` is frozen, or the program goes on to use `attrs` for something else later.

```
Exception: RuntimeError: can't modify frozen Array
--
 0: activerecord-import-0.10.0/lib/activerecord-import/import.rb:495:in `block in add_special_rails_stamps'
 1: activerecord-import-0.10.0/lib/activerecord-import/import.rb:488:in `each_pair'
 2: activerecord-import-0.10.0/lib/activerecord-import/import.rb:488:in `add_special_rails_stamps'
 3: activerecord-import-0.10.0/lib/activerecord-import/import.rb:300:in `import_helper'
 4: activerecord-import-0.10.0/lib/activerecord-import/import.rb:248:in `import'
```

**Comment by indirect** [comment]
(Oops, I left out my suggestion for how to fix it, which is to just call `dup` on the attrs array before you add things to it. HTH!)

**Comment by jbielick** [comment]
+1

**Comment by zdennis** [comment]
Thanks @indirect and @jbielick. This will be resolved shortly in PR #224.

**Comment by joshuamcginnis** [comment]
I just ran into this issue on the current release.

```ruby
columns_to_update = %i[promo_code name desc].freeze

MyTable.import(records, on_duplicate_key_update: {
  conflict_target: [:promo_code],
  columns: columns_to_update
})
``

Raises: 'Exception: RuntimeError: can't modify frozen Array`

**Comment by jkowens** [comment]
@joshuamcginnis good catch, I'm working on an update to make sure on_duplicate_key_update arguments aren't modified.

**Comment by jkowens** [comment]
A fix has been merged into `master` and will be included in the next release.

**Comment by jkowens** [comment]
Just an update, the fix to prevent :on_duplicate_key_update args from being modified has been released in v0.20.1.
